six/modules/c++/six/include/six/ReadControl.h

```diff
@@ -89,19 +89,17 @@ class ReadControl
                       const std::vector<std::string>& schemaPaths) = 0;
 
     /*!
-     *  Get a const pointer to the current container.  This
-     *  container is not allocated, nor deleted by this class
+     *  Get a const pointer to the current container.
      */
-    const Container* getContainer() const
+    mem::SharedPtr<const Container> getContainer() const
```

Comment:
Is the Solaris compiler smart enough to convert from a non-const shared pointer to a const shared pointer?  Sometimes it's dumb about stuff like that.